src/build/app/settings.rs

```diff
@@ -436,6 +439,19 @@ pub enum AppSettings {
     /// [`Arg::default_value`]: ./struct.Arg.html#method.default_value
     ArgRequiredElseHelp,
 
+    /// Instructs the parser to stop when encountering a subcommand instead of greedily consuming
+    /// args.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::{App, AppSettings};
+    /// App::new("myprog")
+    ///     .setting(AppSettings::SubcommandPrecedenceOverArg)
+    /// # ;
+    /// ```
```

Comment:
This example is useless. Could you please copy the code from your test here? And add this explanation:
~~~
By default, if an option taking multiple arguments is followed by a subcommand, the subcommand will be parsed as yet another of the values.
```
app --foo val1 val2 subcommand
          --------- ----------
            values   another value
```

Instructs the parser to stop when encountering a subcommand instead of greedily consuming arguments.
```
app --foo val1 val2 subcommand
          --------- ----------
            values   subcommand
```